tests/gen_cert_test.py

```diff
@@ -181,3 +216,70 @@ def test_cert_date(self, force_date, locale, timezone, expected):
         """
         cert_date = get_cert_date(force_date, locale, timezone)
         assert_equal(expected, cert_date)
+
+
+class CertificateSubtemplateTests(unittest.TestCase):
+    """
+    Tests the subtemplates feature for certs
+
+    Subtemplates are currently written to key on designations only
+
+    These tests have a specific order but do not need to be run in order.
+    They were split and ordered to more easily identify errors while testing.
+    """
+    subtemplate_courses = [
+        'Test/Subtemplates/ByDesignations',
+    ]
+
+    def test_subtemplate_courses_exist(self):
+        """
+        Check that the subtemplate courses exist in settings.CERT_DATA
+        """
+        found_sub_course_ids = []
```

Comment:
yes, thank you.  This was very helpful and to the point.   I should read more about list comprehension and make an effort to use it. 

I'm glad we've move beyond the comments!  I'll need to finish the review later however.